src/app/home/pages/TokensPage/TokensPage.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Card,
+  PageSection,
+  TextContent,
+  Text,
+  CardBody,
+  EmptyState,
+  EmptyStateIcon,
+  Title,
+} from '@patternfly/react-core';
+import { WrenchIcon } from '@patternfly/react-icons';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import clusterSelectors from '../../../cluster/duck/selectors';
+import TokensTable from './components/TokensTable';
+import { useOpenModal } from '../../duck/hooks';
+import { IReduxState } from '../../../../reducers';
+import { IToken } from '../../../token/duck/types';
+import { ICluster } from '../../../cluster/duck/types';
+
+interface ITokensPageBaseProps {
+  tokenList: IToken[];
+  clusterList: ICluster[];
+}
+
+const TokensPageBase: React.FunctionComponent<ITokensPageBaseProps> = ({
+  tokenList,
+  clusterList,
+}: ITokensPageBaseProps) => {
+  const [isAddEditModalOpen, reallyToggleAddEditModal] = useOpenModal(false);
```

Comment:
What's the motivation for the name `reallyToggleX`?

Reply:
That's just a placeholder for this PR, I'm removing it in the next PR. I just didn't want to modify the code below that consumes `toggleAddEditModal` but I wanted to throw an `alert()` inline in it.